subsys/bluetooth/host/classic/a2dp.c

```diff
@@ -703,40 +633,95 @@ int bt_a2dp_stream_config(struct bt_a2dp *a2dp, struct bt_a2dp_stream *stream,
 	stream->a2dp = a2dp;
 	local_ep->stream = stream;
 	remote_ep->stream = stream;
-	a2dp->set_config_param.req.func = bt_a2dp_set_config_cb;
-	a2dp->set_config_param.acp_stream_ep_id = remote_ep->sep.sep_info.id;
-	a2dp->set_config_param.int_stream_endpoint_id = local_ep->sep.sep_info.id;
-	a2dp->set_config_param.media_type = BT_AVDTP_AUDIO;
-	a2dp->set_config_param.media_codec_type = local_ep->codec_type;
-	a2dp->set_config_param.codec_specific_ie_len = config->codec_config->len;
-	a2dp->set_config_param.codec_specific_ie = &config->codec_config->codec_ie[0];
-	a2dp->set_config_param.sep = &local_ep->sep;
+	bt_a2dp_stream_config_set_param(a2dp, config, bt_a2dp_set_config_cb,
+					remote_ep->sep.sep_info.id, local_ep->sep.sep_info.id,
+					local_ep->codec_type, &local_ep->sep);
 	return bt_avdtp_set_configuration(&a2dp->session, &a2dp->set_config_param);
 }
 
-int bt_a2dp_stream_establish(struct bt_a2dp_stream *stream)
+typedef void (*bt_a2dp_rsp_cb)(struct bt_a2dp_stream *stream, uint8_t rsp_err_code);
+typedef void (*bt_a2dp_done_cb)(struct bt_a2dp_stream *stream);
+
+static int bt_a2dp_ctrl_cb(struct bt_avdtp_req *req, bt_a2dp_rsp_cb rsp_cb, bt_a2dp_done_cb done_cb,
+			   bool cler_stream)
```

Comment:
Is it `clear_stream` instead of `cler_stream`?

Reply:
Updated.